### PR TITLE
spec: rich status

### PR DESCRIPTION
**Pull request by xiang90**
/cc @hongchaodeng

### CODE REVIEW

File: pkg/spec/cluster.go [hunk 1]
```diff
@@ -92,9 +93,49 @@ func (c *ClusterSpec) Validate() error {
...
+	ClusterPhaseRunning = "Running"
+	ClusterPhaseFailed  = "Failed"
+)
+
+type ClusterCondition struct {
+	Type ClusterConditionType `json:"type"`
+
+	Reason string `json:"reason"`
+
+	TransitionTime time.Time `json:"transitionTime"`
```
Comment (hongchaodeng) [inline pkg/spec/cluster.go]: What's TransitionTime?

File: pkg/spec/cluster.go [hunk 2]
```diff
@@ -92,9 +93,49 @@ func (c *ClusterSpec) Validate() error {
...
+	ClusterConditionUpgrading = "Upgrading"
+)
+
 type ClusterStatus struct {
+	// Phase is the cluster running phase
+	Phase  ClusterPhase `json:"phase"`
+	Reason string       `json:"reason"`
+
+	// Condition keeps ten most recent cluster conditions
+	Conditions []ClusterCondition `json:"conditions"`
```
Comment (hongchaodeng) [inline pkg/spec/cluster.go]: Why do we need to report an array? What is the array of cond used for?

Comment (xiang90) [inline pkg/spec/cluster.go]: We want to keep a list of conditions.

When user changes size from 3 to 5.

The cluster will turn from condition Ready to ScalingUp. After the cluster has 5 member, the condition will turn from ScalingUp back to Ready.

This is a common practice in k8s world.

File: pkg/spec/cluster.go (hunk 1 above)
Comment (xiang90) [inline pkg/spec/cluster.go]: When the cluster transited to this condition.

**Comment by hongchaodeng** [comment]
lgtm

**Pull request merged**